core/string/string_name.h

```diff
@@ -188,8 +188,22 @@ class StringName {
 	};
 
 	StringName &operator=(const StringName &p_name);
+	StringName &operator=(StringName &&p_name) {
+		if (_data == p_name._data) {
+			return *this;
+		}
+
+		unref();
+		_data = p_name._data;
+		p_name._data = nullptr;
+		return *this;
```

Comment:
Maybe assign nullptr to p_name in both cases ? Or else the caller is still valid after the call, which is fine by move standard, but could be error prone.

Reply:
`nullptr` cannot be assigned to the caller because the `if (_data != p_name._data) {` test covers `this == &p_name` - so we could accidentally be setting our own pointer to `nullptr` accidentally. Besides, if we set its data to `nullptr` without taking ownership of it, we would need to `unref` it.
Regarding your code change suggestion, like last time I still prefer early exit to indented blocks :)

Reply:
That's style anyway, so any will do, but you already have the indented block :). This just remove one line.

Reply:
Yeah - just to clear it up i prefer indented blocks to be the _trivial path_ and indented blocks to be the _important path_. This avoids double indentations and allows you to employ this same principle throughout all of your code, so you always know which is the important path and which the early exit.

Reply:
That's the general principle we go by, early exit is preferred